conans/client/installer.py

```diff
@@ -414,12 +414,12 @@ def _call_package_info(self, conanfile, package_folder, is_editable):
                     # cppinfos inside
 
                     # convert directory entries to be relative to the declared folders.build
-                    build_cppinfo = conanfile.cpp.build.copy()
+                    build_cppinfo = conanfile.cpp.build
```

Comment:
This copy() was shallow, so modifications (making paths absolute with ``set_relative_base_folder``) was still mutating the dependency ``cpp.build`` object.

Reply:
Was it a bug? Was it simply working because we actually wanted to change the dependency `cpp.build` object?